presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java

```diff
@@ -206,6 +210,13 @@ public void testIsDistinctFrom()
         assertOptimizedEquals("9876543210.9874561203 is distinct from NULL", "true");
         assertOptimizedEquals("bound_decimal_short is distinct from NULL", "true");
         assertOptimizedEquals("bound_decimal_long is distinct from 12345678901234567890.123", "false");
+        assertOptimizedMatches("unbound_integer is distinct from 1", "unbound_integer is distinct from 1");
+
+        assertEquals(
```

Comment:
why you cannot use `assertOptimizedMatches` here?

Reply:
The optimized expression contains a Cast that is tagged as "type-only" but the parsed expression doesn't (it doesn't go through analysis and "planning"), so they don't equal each other.